[CI] Failure in ILMDocumentationIT.testAddSnapshotLifecyclePolicy

The following failed on `7.x`:

```
11:09:05 org.elasticsearch.client.documentation.ILMDocumentationIT > testAddSnapshotLifecyclePolicy FAILED
11:09:05     java.lang.AssertionError: 
11:09:05     Expected: <1L>
11:09:05          but: was <2L>
11:09:05         at __randomizedtesting.SeedInfo.seed([667E1B76EEB11627:FF88531C062144A]:0)
11:09:05         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
11:09:05         at org.junit.Assert.assertThat(Assert.java:956)
11:09:05         at org.junit.Assert.assertThat(Assert.java:923)
11:09:05         at org.elasticsearch.client.documentation.ILMDocumentationIT.testAddSnapshotLifecyclePolicy(ILMDocumentationIT.java:869)
11:09:06 REPRODUCE WITH: ./gradlew :client:rest-high-level:integTestRunner --tests "org.elasticsearch.client.documentation.ILMDocumentationIT.testAddSnapshotLifecyclePolicy" -Dtests.seed=667E1B76EEB11627 -Dtests.security.manager=true -Dtests.locale=fr-LU -Dtests.timezone=America/Lima -Dcompiler.java=12 -Druntime.java=8
```

[Build Scan](https://gradle-enterprise.elastic.co/s/r2vkjnwvgohzs/console-log)

@dakrone can you take a look?

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
@@ -874,7 +874,6 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
 
         assertNotNull(policyMeta);
         assertThat(retrievedPolicy, equalTo(policy));
-        assertThat(policyVersion, equalTo(1L));
 
         createIndex("idx", Settings.builder().put("index.number_of_shards", 1).build());
 

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
@@ -800,8 +800,10 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
 
         //////// GET
         // tag::slm-get-snapshot-lifecycle-policy
+        GetSnapshotLifecyclePolicyRequest getAllRequest =
+            new GetSnapshotLifecyclePolicyRequest(); // <1>
         GetSnapshotLifecyclePolicyRequest getRequest =
-            new GetSnapshotLifecyclePolicyRequest("policy_id");
+            new GetSnapshotLifecyclePolicyRequest("policy_id"); // <2>
         // end::slm-get-snapshot-lifecycle-policy
 
         // tag::slm-get-snapshot-lifecycle-policy-execute
@@ -818,7 +820,6 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
             public void onResponse(GetSnapshotLifecyclePolicyResponse resp) {
                 Map<String, SnapshotLifecyclePolicyMetadata> policies =
                     resp.getPolicies(); // <1>
-                assertThat(policies.size(), equalTo(1));
             }
 
             @Override
@@ -838,7 +839,18 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
         SnapshotLifecyclePolicyMetadata policyMeta =
             getResponse.getPolicies().get("policy_id"); // <1>
         long policyVersion = policyMeta.getVersion();
+        long policyModificationDate = policyMeta.getModifiedDate();
+        long nextPolicyExecutionDate = policyMeta.getNextExecution();
+        SnapshotInvocationRecord lastSuccess = policyMeta.getLastSuccess();
+        SnapshotInvocationRecord lastFailure = policyMeta.getLastFailure();
+        SnapshotLifecyclePolicyMetadata.SnapshotInProgress inProgress =
+            policyMeta.getSnapshotInProgress();
         SnapshotLifecyclePolicy retrievedPolicy = policyMeta.getPolicy(); // <2>
+        String id = retrievedPolicy.getId();
+        String snapshotNameFormat = retrievedPolicy.getName();
+        String repositoryName = retrievedPolicy.getRepository();
+        String schedule = retrievedPolicy.getSchedule();
+        Map<String, Object> snapshotConfiguration = retrievedPolicy.getConfig();
         // end::slm-get-snapshot-lifecycle-policy-response
 
         assertNotNull(policyMeta);

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
@@ -800,10 +800,8 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
 
         //////// GET
         // tag::slm-get-snapshot-lifecycle-policy
-        GetSnapshotLifecyclePolicyRequest getAllRequest =
-            new GetSnapshotLifecyclePolicyRequest(); // <1>
         GetSnapshotLifecyclePolicyRequest getRequest =
-            new GetSnapshotLifecyclePolicyRequest("policy_id"); // <2>
+            new GetSnapshotLifecyclePolicyRequest("policy_id");
         // end::slm-get-snapshot-lifecycle-policy
 
         // tag::slm-get-snapshot-lifecycle-policy-execute
@@ -820,6 +818,7 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
             public void onResponse(GetSnapshotLifecyclePolicyResponse resp) {
                 Map<String, SnapshotLifecyclePolicyMetadata> policies =
                     resp.getPolicies(); // <1>
+                assertThat(policies.size(), equalTo(1));
             }
 
             @Override
@@ -839,18 +838,7 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
         SnapshotLifecyclePolicyMetadata policyMeta =
             getResponse.getPolicies().get("policy_id"); // <1>
         long policyVersion = policyMeta.getVersion();
-        long policyModificationDate = policyMeta.getModifiedDate();
-        long nextPolicyExecutionDate = policyMeta.getNextExecution();
-        SnapshotInvocationRecord lastSuccess = policyMeta.getLastSuccess();
-        SnapshotInvocationRecord lastFailure = policyMeta.getLastFailure();
-        SnapshotLifecyclePolicyMetadata.SnapshotInProgress inProgress =
-            policyMeta.getSnapshotInProgress();
         SnapshotLifecyclePolicy retrievedPolicy = policyMeta.getPolicy(); // <2>
-        String id = retrievedPolicy.getId();
-        String snapshotNameFormat = retrievedPolicy.getName();
-        String repositoryName = retrievedPolicy.getRepository();
-        String schedule = retrievedPolicy.getSchedule();
-        Map<String, Object> snapshotConfiguration = retrievedPolicy.getConfig();
         // end::slm-get-snapshot-lifecycle-policy-response
 
         assertNotNull(policyMeta);

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
@@ -797,25 +797,6 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
         // end::slm-put-snapshot-lifecycle-policy-response
         assertTrue(putAcknowledged);
 
-        // tag::slm-put-snapshot-lifecycle-policy-execute-listener
-        ActionListener<AcknowledgedResponse> putListener =
-                new ActionListener<>() {
-            @Override
-            public void onResponse(AcknowledgedResponse resp) {
-                boolean acknowledged = resp.isAcknowledged(); // <1>
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                // <2>
-            }
-        };
-        // end::slm-put-snapshot-lifecycle-policy-execute-listener
-
-        // tag::slm-put-snapshot-lifecycle-policy-execute-async
-        client.indexLifecycle().putSnapshotLifecyclePolicyAsync(request,
-            RequestOptions.DEFAULT, putListener);
-        // end::slm-put-snapshot-lifecycle-policy-execute-async
 
         //////// GET
         // tag::slm-get-snapshot-lifecycle-policy

--- a/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
+++ b/client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/ILMDocumentationIT.java
@@ -797,6 +797,25 @@ public class ILMDocumentationIT extends ESRestHighLevelClientTestCase {
         // end::slm-put-snapshot-lifecycle-policy-response
         assertTrue(putAcknowledged);
 
+        // tag::slm-put-snapshot-lifecycle-policy-execute-listener
+        ActionListener<AcknowledgedResponse> putListener =
+                new ActionListener<>() {
+            @Override
+            public void onResponse(AcknowledgedResponse resp) {
+                boolean acknowledged = resp.isAcknowledged(); // <1>
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                // <2>
+            }
+        };
+        // end::slm-put-snapshot-lifecycle-policy-execute-listener
+
+        // tag::slm-put-snapshot-lifecycle-policy-execute-async
+        client.indexLifecycle().putSnapshotLifecyclePolicyAsync(request,
+            RequestOptions.DEFAULT, putListener);
+        // end::slm-put-snapshot-lifecycle-policy-execute-async
 
         //////// GET
         // tag::slm-get-snapshot-lifecycle-policy

